src/haxeLanguageServer/features/haxe/DiagnosticsFeature.hx

```diff
@@ -183,8 +180,7 @@ class DiagnosticsFeature {
 				};
 				final diag:Diagnostic = {
 					range: range,
-					source: DiagnosticsSource,
-					code: kind,
+					code: hxDiag.code,
```

Comment:
my tests with nightly and 4.3.x always had `hxDiag.code` as `null` and since `kind` isn't preserved (or used as `code`) our diagnostic actions fail to contribute any quickfixes.

so how is the new `code` system supposed to work? and what do we need to adjust to fix our code actions?

Reply:
Need https://github.com/HaxeFoundation/haxe/pull/11207 to get the code from Haxe (which wouldn't give the values you expect)
Using code to decide what quickfix to add sounds hacky, can't/shouldn't tags be used?

Reply:
currently `DiagnosticTag` has two values: `Unnecessary` and `Deprecated`. technically it's an Int and we could cast any number to it, and it might work. 
we probably should take some higher values for our tags though, to avoid potential conflict with future extensions of that enum on vscode's side.

Reply:
I should take a look at what we actually used that code (well, kind) for and see how I could "un-break" it without breaking the rest of the features I was working on

Reply:
Is there more than `DiagnosticsCodeActionFeature` to handle?

Reply:
Ok I see several more using `context.diagnostics.getArguments()` which may still be working, but I'll have to check those too.

Reply:
not that I'm aware of

Reply:
Noticed that https://github.com/vshaxe/haxe-language-server/blob/master/src/haxeLanguageServer/features/haxe/codeAction/diagnostics/CompilerErrorActions.hx#L36 can't work anymore since that commit 4 years ago x)
https://github.com/HaxeFoundation/haxe/commit/d415a7d8f18a08c6e253e38057b165392173d6dc